rest/test_results_routes.go

```diff
@@ -67,6 +77,50 @@ func (h *testResultsGetByTaskIDHandler) Run(ctx context.Context) gimlet.Responde
 	return gimlet.NewJSONResponse(testResults)
 }
 
+///////////////////////////////////////////////////////////////////////////////
+//
+// GET /test_results/task_id/{task_id}/failed_sample
+
+type testResultsGetFailedSampleHandler struct {
+	sc data.Connector
+	testResultsBaseHandler
+}
+
+func makeGetTestResultsFailedSample(sc data.Connector) gimlet.RouteHandler {
+	return &testResultsGetFailedSampleHandler{
+		sc: sc,
+	}
+}
+
+// Factory returns a pointer to a new testResultsGetFailedSampleHandler.
+func (h *testResultsGetFailedSampleHandler) Factory() gimlet.RouteHandler {
+	return &testResultsGetFailedSampleHandler{
+		sc: h.sc,
+	}
+}
+
+// Run finds and returns the desired failed test results sample.
+func (h *testResultsGetFailedSampleHandler) Run(ctx context.Context) gimlet.Responder {
+	sample, err := h.sc.FindFailedTestResultsSample(ctx, h.opts)
+	if err != nil {
+		taskID := h.opts.TaskID
+		if taskID == "" {
+			taskID = h.opts.DisplayTaskID
+		}
+		err = errors.Wrapf(err, "problem getting test result by task_id '%s'", taskID)
+		grip.Error(message.WrapError(err, message.Fields{
+			"request":      gimlet.GetRequestID(ctx),
+			"method":       "GET",
+			"route":        "/testresults/task_id/{task_id}/sample",
```

Comment:
Should the route end in `/failed_sample`?

Reply:
Good catch, updated.